### ISSUE TITLE
Not workimg with QEMU emulator version 2.11.0 at all..

### ISSUE THREAD

**Issue by thpryrchn** [pmj/virtio-net-osx#10]
The device seems to be a newer version than the driver is written for, and even has checks against it....

Here is my ioreg for my virtio device..
https://www.pastiebin.com/5a4c5c1a1a889

I am running on Sierra on Arch Linux, which I guess has the latest QEMU....

**Comment by pmj** [comment]
Yup, looks like 2.11 now implements Virtio 1.0 standard, non-transitional devices. Previous versions have been using transitional devices, which have implemented both the 1.0 standard device interface, and the legacy interface which the original virtio-net-osx driver targets. I actually started working on a new, Virtio PCI 1.0 backend just before Christmas. It's not quite there yet, but I'm hoping I can keep working on it through January despite work commitments.

Until that work is done, you may be able to revert Qemu to the old behaviour. [This documentation](https://github.com/qemu/qemu/blob/master/docs/pcie.txt) suggests that setting the following option in the qemu command line argument for the virtio network device might turn the legacy support back on:

    disable-legacy=off

Note that the old driver also doesn't support MSI-X, so you'll need to set the vectors option to 0 as well if you have trouble.

I haven't tested the above workaround, sorry - let me know if it doesn't work and we'll try to find a better alternative.

**Comment by pmj** [comment]
P.S. Please keep this issue open until it's properly fixed and I release 1.0b1, even if the workaround solves it for now.

**Comment by thpryrchn** [comment]
Hmmm. I have not figured out how to do that with the libvirt XML that I have my Mac machine in... I may have to try converting it to a command line, and see if it will work.

**Comment by thpryrchn** [comment]
If you need any testing, Let me know :)

**Comment by pmj** [comment]
It will definitely need a substantial amount of testing, but the standard-compliant virtio PCI backend driver is not quite functional yet, I'm afraid, so not ready for anything practical just yet. I'll definitely let you know when that changes.

**Comment by steeve** [comment]
hey @pmj, any news about 1.0b1 ?
On a side note, this is truly amazing work
